src/lib/x509/certstor_system_windows/certstor_windows.cpp

```diff
@@ -0,0 +1,236 @@
+/*
+* Certificate Store
+* (C) 1999-2019 Jack Lloyd
+* (C) 2018-2019 Patrik Fiedler, Tim Oesterreich
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/certstor_windows.h>
+#include <botan/der_enc.h>
+
+#include <array>
+#include <vector>
+#include <iostream>
+
+#define NOMINMAX 1
+#define _WINSOCKAPI_ // stop windows.h including winsock.h
+#include <Windows.h>
+#include <Wincrypt.h>
+
+namespace {
+
+const std::array<std::string, 4> cert_store_names{"MY", "Root", "Trust", "CA"};
```

Comment:
Does it make sense to hardcode these? They can (AIUI) change depending on configuration. They can be enumerated with CertEnumSystemStore

Reply:
Assuming we only want to find trust roots, it seems to be sufficient to search `Root` and `CA`. According to the docs, `MY` contains certificates of the current user with associated private keys.

I don't feel confident enough to use `CertEnumSystemStore` correctly... AFAICS it shouldn't be necessary.